dbms/src/Storages/KVStore/KVStore.cpp

```diff
@@ -483,4 +495,126 @@ RegionTaskLock KVStore::genRegionTaskLock(UInt64 region_id) const
     return region_manager.genRegionTaskLock(region_id);
 }
 
+static std::string getThreadNameAggPrefix(const std::string_view & s)
+{
+    if (auto pos = s.find_last_of('-'); pos != std::string::npos)
+    {
+        return std::string(s.begin(), s.begin() + pos);
+    }
+    return std::string(s.begin(), s.end());
+}
+
+void KVStore::reportThreadAllocInfo(std::string_view thdname, ReportThreadAllocateInfoType type, uint64_t value)
+{
+    // Many threads have empty name, better just not handle.
+    if (thdname.empty())
+        return;
+    std::string tname(thdname.begin(), thdname.end());
+    switch (type)
+    {
+    case ReportThreadAllocateInfoType::Reset:
+    {
+        {
+            auto & metrics = TiFlashMetrics::instance();
+            std::unique_lock lock(metrics.proxy_thread_ru_mtx);
+            if unlikely (!metrics.registered_raft_proxy_thread_memory_usage_metrics.count(k))
+            {
+                // Add new keyspace store usage metric
+                metrics.registered_raft_proxy_thread_memory_usage_metrics.emplace(
+                    k,
+                    &metrics.registered_raft_proxy_thread_memory_usage_family->Add({{"type", k}}));
+            }
+        }
```

Comment:
wrap it as a method in `TiFlashMetrics` so that we won't easily miss the lock on `proxy_thread_ru_mtx`?

Reply:
fixed